modules/EnsEMBL/Web/SpeciesDefs.pm

```diff
@@ -72,299 +73,39 @@ sub abbreviated_species_label {
-
-    ## Check for a) genome-specific image, b) species-specific image
-    my $no_image  = 1;
-    ## Need to use full path, as image files are usually in another plugin
-    my $image_dir = sprintf '%s/eg-web-%s/htdocs/i/species', $SiteDefs::ENSEMBL_SERVERROOT, $SiteDefs::DIVISION;
-    my $image_path = sprintf '%s/%s.png', $image_dir, $key;
-    ## In theory this could be a trinomial, but we use whatever is set in species.species_name
-    my $binomial = $tree->{$key}{SPECIES_BINOMIAL};
-    if ($binomial) {
-      $binomial =~ s/ /_/g;
-    }
-    else {
-      ## Make a guess based on URL. Note that some fungi have weird URLs bc 
-      ## their taxonomy is uncertain, so this regex doesn't try to include them
-      ## (none of them have images in any case)
-      $key =~ /^([A-Za-z]+_[a-z]+)/;
-      $binomial = $1;
-    }
-    my $species_path = $binomial ? sprintf '%s/%s.png', $image_dir, $binomial : '';
-    if (-e $image_path) {
-      $tree->{$key}{'SPECIES_IMAGE'} = $key;
-      $no_image = 0;
-    }
-    elsif ($species_path && -e $species_path) {
-      $tree->{$key}{'SPECIES_IMAGE'} = $binomial;
-      $no_image = 0;
-    }
-    else {
-      $tree->{$key}{'SPECIES_IMAGE'} = 'default';
-      $no_image = 0;
-    }
-  }
-
-  $tree->{'MULTI'}{'ENSEMBL_DATASETS'} = $datasets;
-  #warn ">>> NEW KEYS: ".Dumper($tree);
-
-  ## File format info 
-  my $format_info = $self->_get_file_format_info($tree);;
-  $tree->{'MULTI'}{'UPLOAD_FILE_FORMATS'} = $format_info->{'upload'};
-  $tree->{'MULTI'}{'REMOTE_FILE_FORMATS'} = $format_info->{'remote'};
-  $tree->{'MULTI'}{'DATA_FORMAT_INFO'} = $format_info->{'formats'};
-
-  ## Parse species directories for static content
-  $tree->{'SPECIES_INFO'} = $self->_load_in_species_pages;
-  {
-    no strict "vars";
-    $CONF->{'_storage'} = $tree; # Store the tree
-  }
-  $self->_info_line('Filesystem', 'Trawled species static content');
-
+  my $cow_from_defaults = { 
+                            'ENSEMBL_SPECIES_SITE'  => 1,
+                            'SPECIES_DISPLAY_NAME'  => 1 
+                          };
+  $cow_from_defaults->{'ENSEMBL_EXTERNAL_URLS'} = 1 if $SiteDefs::ENSEMBL_SITETYPE =~ /bacteria/i;
+  return $cow_from_defaults;
 }
 
-our %cow_from_defaults = ( # copy-on-write from defautls for these sections.
-                          'ENSEMBL_SPECIES_SITE'  => 1,
-                          'SPECIES_DISPLAY_NAME'  => 1 );
-
-$cow_from_defaults{'ENSEMBL_EXTERNAL_URLS'} = 1 if $SiteDefs::ENSEMBL_SITETYPE =~ /bacteria/i;
-
 sub _merge_db_tree {
   my ($self, $tree, $db_tree, $key) = @_;
   return unless defined $db_tree;
   Hash::Merge::set_behavior('RIGHT_PRECEDENT');
   my $t = merge($tree->{$key}, $db_tree->{$key});
-  foreach my $k ( %cow_from_defaults ) {
+  my %cow_from_defaults = %{$self->_get_cow_defaults};
+  foreach my $k ( keys %cow_from_defaults ) {
       $t->{$k} = $tree->{$key}->{$k} if defined $tree->{$key}->{$k};
```

Comment:
I know this is no update on this line, but noticed should we also check if $tree->{$key} is defined here?

Reply:
I'd rather not touch this code - the config merging is pretty arcane, and it's been working fine for years!

Reply:
Loll ok fine :-)